File: ordertracker-backend/src/main/java/hu/unideb/inf/ordertrackerbackend/api/model/Order.java
package hu.unideb.inf.ordertrackerbackend.api.model;

import hu.unideb.inf.ordertrackerbackend.api.enums.OrderStatus;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Set;

@Entity
@Table(name = "orders")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long userId;

    private OrderStatus status;

    private LocalDateTime createdAt = LocalDateTime.now();

    private LocalDateTime statusLastModify = LocalDateTime.now();

    @Transient
    private float sumOfItems;

    @OneToMany(mappedBy = "order")
    private Set<OrderItem> orderItems;

    public void addOrderItem(OrderItem orderItem) {
        this.orderItems.add(orderItem);
    }

    public Order(OrderStatus status) {
        this.status = status;
    }

    public void calculateSumOfItems() {
        float sum = 0f;

        for (OrderItem orderItem : orderItems) {
            sum += orderItem.getPrice() * orderItem.getCount();
        }

        sumOfItems = sum;
    }

    public void removeOrderItems() {
        orderItems = null;
    }

}

File: ordertracker-android/app/src/main/java/hu/unideb/inf/ordertrackerandroid/model/api/Order.kt
package hu.unideb.inf.ordertrackerandroid.model.api

import hu.unideb.inf.ordertrackerandroid.network.OrderStatus
import java.time.LocalDateTime

class Order {
    val id: Long? = null
    val userId: Long? = null
    val status: OrderStatus? = null
    val orderItems: Set<OrderItem>? = null
    val createdAt: LocalDateTime? = null
    val statusLastModify: LocalDateTime? = null
    val sumOfItems: Float? = null
}